src/main/java/org/terasology/climateConditions/BodyTemperatureSystem.java

```diff
@@ -0,0 +1,107 @@
+import org.terasology.registry.In;
+import org.terasology.world.WorldComponent;
+
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class BodyTemperatureSystem extends BaseComponentSystem {
+    public static final String BODY_TEMPERATURE_UPDATE_ACTION_ID = "Body Temperature Update";
+    public static final String BODY_TEMPERATURE_DISPLAY_ACTION_ID = "Body Temperature Display";
+
+    private static final Logger logger = LoggerFactory.getLogger(BodyTemperatureSystem.class);
+
+    @In
+    private EntityManager entityManager;
+    @In
+    ClimateConditionsSystem climateConditionsSystem;
+    @In
+    DelayManager delayManager;
+
+    private static final int CHECK_INTERVAL = 1000;
+    private static final int DISPLAY_INTERVAL = 10000;
+
+    public void postBegin() {
+        boolean processedOnce = false;
+        for (EntityRef entity : entityManager.getEntitiesWith(WorldComponent.class)) {
+            if (!processedOnce) {
+                delayManager.addPeriodicAction(entity, BODY_TEMPERATURE_UPDATE_ACTION_ID, 0, CHECK_INTERVAL);
+                processedOnce = true;
+            } else {
+                logger.warn("More than one entity with WorldComponent found");
+            }
+        }
+        // The world is the entity here which shall have a Body Temperature Update Periodic Action.
+    }
+
+    @ReceiveEvent
+    public void onTemperatureUpdate(PeriodicActionTriggeredEvent event, EntityRef world) {
+        if (event.getActionId().equals(BODY_TEMPERATURE_UPDATE_ACTION_ID)) {
+            for (EntityRef entity : entityManager.getEntitiesWith(AliveCharacterComponent.class)) {
+                LocationComponent location = entity.getComponent(LocationComponent.class);
+                if (!entity.hasComponent(BodyTemperatureComponent.class)) {
+                    BodyTemperatureComponent btc = new BodyTemperatureComponent();
+                    entity.addComponent(btc);
+                }
+                BodyTemperatureComponent btc = entity.getComponent(BodyTemperatureComponent.class);
+                float envTemperature = climateConditionsSystem.getTemperature(location.getLocalPosition().getX(),
+                        location.getLocalPosition().getY(), location.getLocalPosition().getZ());
+                float envHumidity = climateConditionsSystem.getHumidity(location.getLocalPosition().getX(),
+                        location.getLocalPosition().getY(), location.getLocalPosition().getZ());
+                btc.bodyTemperature =
+                        btc.lastBodyTemperature + ((((envTemperature - (envHumidity / 10)) - btc.lastBodyTemperature) / 100000) * CHECK_INTERVAL);
+                btc.lastBodyTemperature = btc.bodyTemperature;
+                entity.saveComponent(btc);
+            }
+        }
+    }
+
+    @ReceiveEvent
+    public void onTemperatureDisplay(PeriodicActionTriggeredEvent event, EntityRef player, LocationComponent location
+            , BodyTemperatureComponent btc) {
```

Comment:
formatting looks a bit off

Reply:
I know, but when I changed it, then again used ctrl+alt+l , it changed back to this.

Reply:
oh no! If you can find out which setting it is that thinks it should use that leading-comma style, I think we would all be happy to turn it off.

I think I've looked for it before. It seems like it must be under Editor / Code Style / Java, but I haven't found anything in there that mentions commas.

Reply:
huh weird